main.go

```diff
@@ -149,12 +148,27 @@ func main() {
 	cacheLooper := director.NewTimedLooper(
 		director.FOREVER, config.CacheFlushInterval, make(chan error))
 
-	// In the event our filter can't find the right creds, etc, we fail open
-	if podFilter != nil {
-		filter = podFilter
+	// Choose the appropriate filter implementation
+	if config.TailAll {
+		tailAllFilter := NewTailAllFilter(
+			config.KubeHost, config.KubePort, config.KubeTimeout, config.KubeCredsPath,
+		)
+		if tailAllFilter != nil {
+			filter = tailAllFilter
+		} else {
+			log.Warn("Failed to configure TailAll filter, proceeding anyway using stub...")
+			filter = &StubFilter{TailAll: true}
+		}
 	} else {
-		log.Warn("Failed to configure filter, proceeding anyway using stub...")
-		filter = &StubFilter{}
+		annotationFilter := NewAnnotationPodFilter(
+			config.KubeHost, config.KubePort, config.KubeTimeout, config.KubeCredsPath,
+		)
+		if annotationFilter != nil {
+			filter = annotationFilter
+		} else {
+			log.Warn("Failed to configure Annotation filter, proceeding anyway using stub...")
+			filter = &StubFilter{TailAll: false}
+		}
```

Comment:
```suggestion
	}
	
	if podFilter == nil {
		log.Warn("Failed to configure Annotation filter, proceeding anyway using stub...")
		podFilter = &StubFilter{TailAll: false}
	}
```

Reply:
Obviously you can use `filter` instead of `podFilter` but you get the idea, I hope.